### PR TITLE
Milestone 3: Users & posts

### PR DESCRIPTION
**Pull request by Luzaks**
In this pull request we create and style:

- The User and Post model with it's associations.

- The Users and Posts controllers with it's views and it's methods.

- Unit tests and integration test.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -1,3 +1,15 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see https://guides.rubyonrails.org/routing.html
+  devise_scope :user do
+    authenticated :user do
+      root to: 'users#show', as: :authenticated_root
+    end
+  
+    unauthenticated :user do
+      root to: 'static_pages#home', as: :unauthenticated_root
```
Comment (mosaaleb) [inline config/routes.rb]: - [ ] **[OPTIONAL]** Nice work have separate routes for authenticated and unauathenticated users, but to get rid of the redundant message 'You need to log in before continuiing', it's better to set the root for unauthenticated to be sign_in page.
```ruby
unauthenticated :user do
    root to: 'devise/sessions#new'
end
```

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
+class PostsController < ApplicationController
+  def create
```
Comment (mosaaleb) [inline app/controllers/posts_controller.rb]: - [ ] PostsController#index displays posts in timeline


Let's add the index page for posts.

File: app/views/posts/_post.html.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<li id="post-<%= post.id %>">
```
Comment (mosaaleb) [inline app/views/posts/_post.html.erb]: - [ ] Post is always displayed with content and author


-- Let's add the post author.

File: app/views/posts/_post.html.erb [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  <span class="content"><%= post.content %></span>
+  <span class="timestamp">
+    Posted <%= time_ago_in_words(post.created_at) %> ago.
+  </span>
+  <%= link_to "delete", post, method: :delete,
```
Comment (mosaaleb) [inline app/views/posts/_post.html.erb]: - [ ] We need to check here first if the current user is the same as post author before displaying the delete button.
Don't forget to move this logic in helper method. [helpers](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers)

File: app/views/users/show.html.erb [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    <div class= "col-md-4 col-sm-12 user-info text-center">
+      <%= gravatar_for @current_user %>
+      <p> <%= @current_user.name %> <%= @current_user.lastname %> </p>
+      <p><%= @current_user.email %> </p>
+    </div>
+
+    <div class="col-md-8 post-info text-center">
+    
+    <%= render 'shared/newpost' %>
+    <% if @current_user.posts.any? %>
```
Comment (mosaaleb) [inline app/views/users/show.html.erb]: - [ ] There shoud be absolutely NO logic in views.
Rails has built-in mechanism to deal with view-related logic which is View Helpers (app/helpers/*_helper.rb). So the common convention is to move any logic in the views to the helpers. Please take a look at [this](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) for more information.

File: app/views/posts/_post.html.erb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+<li id="post-<%= post.id %>">
+  <p class="content"><%= post.user.name %>    <%= post.user.lastname %></p>
```
Comment (samgaco) [inline app/views/posts/_post.html.erb]: :+1:

File: app/models/post.rb [hunk 7]
```diff
@@ -0,0 +1,7 @@
+class Post < ApplicationRecord
+  belongs_to :user, class_name: 'User', foreign_key: 'author'
+  default_scope -> { order(created_at: :desc) }
```
Comment (samgaco) [inline app/models/post.rb]: :+1:

**Pull request merged**